packages/ledger/src/utils/payment.utils.ts

```diff
@@ -0,0 +1,35 @@
+import { isNullish, nonNullish } from "@dfinity/utils";
+
+/**
+ * A naive implementation of a payment parser. Given a code, the function attempts to extract a token name, account identifier (textual representation), and an optional amount.
+ *
+ * If the code doesn't match the expected pattern, `undefined` is returned for simplicity.
+ * Similarly, if an optional amount is provided but it's not a valid number, the parser will not throw an exception and returns `undefined`.
+ *
+ * Please note that this function doesn't perform any validity checks on the extracted information.
+ * It doesn't verify if the token is known or if the identifier is a valid address.
+ *
+ * @param code string
+ * @returns { token: string; identifier: string; amount?: number } | undefined
+ */
+export const decodePayment = (
+  code: string
+): { token: string; identifier: string; amount?: number } | undefined => {
+  const regex =
+    /^([a-zA-Z]+):([A-Za-z0-9:\-.]+).*?(?:[?&](?:amount|value)=(\d+(?:\.\d+)?))?$/;
```

Comment:
Why is there `amount|value`? Are both supported? Why not choosing one?

Reply:
same as above, once we have a proposal will document / link it

Reply:
yes, I saw the tests, I was reconfirming because I was surprised to see both supported.

Thanks for the explanation.